lms/djangoapps/teams/static/teams/js/collections/base.js

```diff
@@ -0,0 +1,23 @@
+;(function (define) {
+    'use strict';
+    define(['common/js/components/collections/paging_collection'],
+        function(PagingCollection) {
+            var BaseCollection = PagingCollection.extend({
+                initialize: function(options) {
+                    PagingCollection.prototype.initialize.call(this);
+
+                    this.course_id = options.course_id;
+                    this.perPage = options.per_page;
+
+                    this.teamEvents = options.teamEvents;
+                    this.teamEvents.bind('teams:update', this.onUpdate, this);
+                    this.isStale = false;
+                },
+
+                onUpdate: function(event) {
+                    this.isStale = true;
+                }
```

Comment:
I'm conflicted because the ``self`` here is the view, not the collection. How about we have a ``refresh`` method on these collections that does the fetch and clears the stale flag, and then the view call:

    self.collection.refresh().done(...)

I'll try that and see how it feels.